### ISSUE TITLE
Fix resource leaks in test_zipstream.py and test_paths.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8569]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8569|
|Type|enhancement|
|Created|2016-07-07 02:52:52Z|
|Branch|https://github.com/twisted/twisted/tree/8569-rodrigc-pypy|

test_zipstream.py and test_paths.py have some resource leaks
(open file descriptors).  These resource leaks cause errors when
the tests are run under Pypy.

<details><summary>Searchable metadata</summary>

```
trac-id__8569 8569
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__PyPy_support PyPy-support
branch__8569_rodrigc_pypy 8569-rodrigc-pypy
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1467859972083662 1467859972083662
changetime__1468035956558097 1468035956558097
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/298

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_2"></a>|@markrwilliams commented|
|-|-|

I've added my review notes to the PR, which seems to need a merge from trunk right now.  

I say this is OK to merge!

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset d952db63d91de74dd1b50bb822895ec309b15292
```
#!CommitTicketReference repository="" revision="d952db63d91de74dd1b50bb822895ec309b15292"
Merge pull request #298 from twisted/8569-rodrigc-pypy

Author: rodrigc
Reviewer: markrwilliams
Fixes: #8569

Plug some file descriptor leaks in test_paths.py and test_zipstream.py
```
